data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# shellcheck disable=SC1091
+. /usr/local/bin/release-image.sh
+
+export KUBECONFIG=/etc/kubernetes/kubeconfig
+
+# Wait till the baremetalhosts are populated
+until oc get baremetalhosts -n openshift-machine-api; do
+   echo Waiting for BareMetalHosts to appear...
+   sleep 20
+done
+
+# Wait for 3 masters to appear.
+while [ "$(curl -s http://localhost:6385/v1/nodes | jq '.nodes[] | .uuid' | wc -l)" -lt 3 ]; do
+    echo waiting for at least 3 masters.
```

Comment:
Sorry, one more thought here: 3 might be a bad assumption. Single master installs have historically been possible and used by developers with minimal hardware. It's unsupported though.

We could maybe count how many BareMetalHost CR's there are with `master` in the name, or read the actual value from /opt/openshift/manifests/cluster-config.yaml.

@abhinavdahiya might have thoughts here.

Reply:
or just check even for 1? If there's 1 it will not be 'active'.

Reply:
I'll test it and see if that works

Reply:
Yea I worried about a race but the next step takes like 10 minutes at least, so the installer will have created all the nodes by then

Reply:
Yes ok.  I tested it with -lt 1, meaning we just need 1 node to come up and it continues.  I also didn't use any sleep in the loop so it queried constantly.  As soon as a node comes up its state is 'enroll', and then moves to introspection states, so I think we are safe looking for 1 node and then on to the next check.